google/cloud/firestore_v1/async_collection.py

```diff
@@ -112,6 +112,23 @@ async def add(
         write_result = await document_ref.create(document_data, **kwargs)
         return write_result.update_time, document_ref
 
+    def document(
+        self, document_id: str = None
+    ) -> async_document.AsyncDocumentReference:
+        """Create a sub-document underneath the current collection.
+
+        Args:
+            document_id (Optional[str]): The document identifier
+                within the current collection. If not provided, will default
+                to a random 20 character string composed of digits,
+                uppercase and lowercase and letters.
+
+        Returns:
+            :class:`~google.cloud.firestore_v1.document.async_document.AsyncDocumentReference`:
+            The child document.
+        """
+        return super(AsyncCollectionReference, self).document(document_id)
```

Comment:
Hmm, it's unfortunate that we still use the Python 2 syntax here, though I see that's the pattern across the library and you were just following it. Nothing to do here except collectively shrug our shoulders.

Reply:
You are welcome to go through and refactor this. This is less a choice and more a fact that this library evolved from Py2/3 compliant code and we didn't refactor more than necessary to use new features really (AsyncIO, f strings, etc)